pkg/templates/machinecontroller/deployment.go

```diff
@@ -715,16 +715,23 @@ func machineControllerMachineDeploymentCRD() *apiextensions.CustomResourceDefini
 func machineControllerDeployment(cluster *kubeoneapi.KubeOneCluster) (*appsv1.Deployment, error) {
 	var replicas int32 = 1
 
-	clusterDNS, err := clusterDNSIP(cluster)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to get clusterDNS IP")
+	var clusterDNSstring string
+
+	if cluster.Features.NodeLocalDNSCache != nil && cluster.Features.NodeLocalDNSCache.Enable {
+		clusterDNSstring = "169.254.20.10"
+	} else {
+		clusterDNS, err := clusterDNSIP(cluster)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get clusterDNS IP")
+		}
+		clusterDNSstring = clusterDNS.String()
 	}
 
 	args := []string{
 		"-logtostderr",
 		"-v", "4",
 		"-internal-listen-address", "0.0.0.0:8085",
-		"-cluster-dns", clusterDNS.String(),
+		"-cluster-dns", clusterDNSstring,
```

Comment:
Do we need to set anything similar on the control plane nodes? As far as I know, this is a kubelet setting, so eventually, we maybe need to have the same value on all kubelets.

Reply:
eventually yes

Reply:
done